### PR TITLE
removed RouteProgressState.ROUTE_INVALID state and renamed the states

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

We already aren't reporting any route progress state when there's no route object, so we also shouldn't report any route progress for other calculation failures - missing route being one of them but also missing location updates, etc.

@mapbox/navnative is there any scenario where we would actually like to deliver `RouteState#INVALID` to developers that we are missing?

This closes https://github.com/mapbox/mapbox-navigation-android/issues/4379 by not emitting a `ROUTE_INVALID` state which defaults to
```
distanceRemaining == 0
durationRemaining == 0
```
https://github.com/mapbox/mapbox-navigation-android/blob/bc1dce0cb8cba9ea8c8efc40bd8d65d5350a9d23/libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt#L160-L162

and was triggering arrival observer: https://github.com/mapbox/mapbox-navigation-android/blob/bc1dce0cb8cba9ea8c8efc40bd8d65d5350a9d23/libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalProgressObserver.kt#L98-L109

We were ending up emitting the `ROUTE_INVALID` because the status updater was started by https://github.com/mapbox/mapbox-navigation-android/blob/bc1dce0cb8cba9ea8c8efc40bd8d65d5350a9d23/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L100

before there were any location updates available.

### Changelog

```
<changelog>Removed `RouteProgressState.ROUTE_INVALID`. `RouteProgress` will never be delivered with an invalid update, besides `RouteProgressState.OFF_ROUTE`.</changelog>
<changelog>Removed the `ROUTE_` prefix from `RouteProgressState` values and updated the documentation.</changelog>
<changelog>Moved the `RouteProgressState.LOCATION_STALE` to `RouteProgress.stale`.</changelog>
```

### CODE REVIEW

**Comment by LukasPaczos** [comment]
This PR now also renames the states and updates the documentation.

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgressState.kt [hunk 1]
```diff
@@ -1,54 +1,38 @@
...
-     * State when we start following a route.
-     *
-     * After a certain number of tracking points, we gain confidence and switch to tracking state.
-     * We do map-matching rather than route line snapping during this state.
+     * A lack of [android.location.Location] updates has caused a lack of confidence in the
+     * progress updates being sent.
      */
-    ROUTE_UNCERTAIN
+    LOCATION_STALE
 }
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgressState.kt:38]: @mapbox/navnative could you double-check that this documentation is correct?

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgressState.kt [hunk 2]
```diff
@@ -1,54 +1,38 @@
...
     /**
-     * [com.mapbox.navigation.core.MapboxNavigation] is now confidently tracking the
-     * location updates and processing them against the route.
-     */
-    LOCATION_TRACKING,
-
-    /**
-     * A lack of [android.location.Location] updates has caused a lack of confidence in the
-     * progress updates being sent.
+     * We've gone of route: all [RouteProgress] fields are invalid.
```
Comment (mskurydin) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgressState.kt:24]: nit: off

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgressState.kt [hunk 3]
```diff
@@ -1,54 +1,38 @@
...
-    OFF_ROUTE,
+    UNCERTAIN,
 
     /**
-     * State when we start following a route.
-     *
-     * After a certain number of tracking points, we gain confidence and switch to tracking state.
-     * We do map-matching rather than route line snapping during this state.
+     * A lack of [android.location.Location] updates has caused a lack of confidence in the
+     * progress updates being sent.
```
Comment (mskurydin) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgressState.kt:35]: we no longer have a `stale` state. The stale sign is propagated through a separate indicator in the `NavigationStatus`

Comment (LukasPaczos) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgressState.kt:35]: right, we bundle it in the progress now. Would you recommend decoupling it?

Comment (LukasPaczos) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgressState.kt:35]: Movde to `RouteProgress.stale`.

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt [hunk 4]
```diff
@@ -153,7 +160,7 @@ class RouteProgress private constructor(
         private var routeGeometryWithBuffer: Geometry? = null
         private var bannerInstructions: BannerInstructions? = null
         private var voiceInstructions: VoiceInstructions? = null
-        private var currentState: RouteProgressState = RouteProgressState.ROUTE_INVALID
+        private var currentState: RouteProgressState = RouteProgressState.INITIALIZED
```
Comment (Guardiola31337) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt:163]: Is it actually `INITIALIZED` at this point? 🤔

**Review by Guardiola31337 — APPROVED** [review]
Left a couple of minor comments 🚀

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt (hunk 4 above)
Comment (LukasPaczos) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt:163]: No, but a nullable `RouteProgressState` wouldn't be ideal either.

I think we should remove the builder pattern and move `RouteProgress` to an internal factory as well - https://github.com/mapbox/mapbox-navigation-android/issues/4414.

**Pull request merged**